R/utilities.R

```diff
@@ -805,181 +805,414 @@ trim_scale_expression = function(x){
 #' @param mode Only works with indexed flat-files. Accepts 2 options of "slms-3" and "strelka2" to indicate which variant caller to use. Default is "slms-3".
 #'
-#' @return Count matrix.
+#' @return Either a matrix or a long tidy table of counts per window.
 #'
 #' @rawNamespace import(data.table, except = c("last", "first", "between", "transpose"))
-#' @import dplyr tidyr cowplot ggplot2
+#' @import dplyr tidyr
 #' @export
 #'
 #' @examples
 #' chr11_mut_freq = calc_mutation_frequency_sliding_windows(this_region = "chr11:69455000-69459900",
 #'                                                          slide_by = 10,
 #'                                                          window_size = 10000)
 #'
-calc_mutation_frequency_sliding_windows = function(this_region,
-                                                   chromosome,
-                                                   start_pos,
-                                                   end_pos,
-                                                   metadata,
-                                                   seq_type,
-                                                   slide_by = 100,
-                                                   window_size = 1000,
-                                                   plot_type = "none",
-                                                   sortByColumns = "pathology",
-                                                   return_format = "long-simple",
-                                                   min_count_per_bin = 3,
-                                                   return_count = FALSE,
-                                                   drop_unmutated = FALSE,
-                                                   classification_column = "lymphgen",
-                                                   from_indexed_flatfile = FALSE,
-                                                   mode = "slms-3"){
-
-  max_region = 1000000
-  if(missing(metadata)){
-    metadata = get_gambl_metadata()
-  }
-  if(missing(this_region)){
-    this_region = paste0(chromosome, ":", start_pos, "-", end_pos)
-  }else{
-    chunks = region_to_chunks(this_region)
-    chromosome = chunks$chromosome
-    start_pos = as.numeric(chunks$start)
-    end_pos = as.numeric(chunks$end)
-  }
-  region_size = end_pos - start_pos
-  if(region_size < max_region){
-    message(paste("processing bins of size", window_size, "across", region_size, "bp region"))
-  }else{
+
+calc_mutation_frequency_sliding_windows <- function(this_region,
+                                          chromosome,
+                                          start_pos,
+                                          end_pos,
+                                          these_samples_metadata,
+                                          these_sample_ids,
+                                          maf = NULL,
```

Comment:
Since some functions work with both, I try to use maf_df for maf data frame/object and maf_file for a path to a file.